third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -1658,6 +1658,36 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         )
     }
 
+    /// This checks whether result_exp contains mutable borrow of a field from an immutable reference
+    /// It needs to be called after `post_process_body`
+    pub fn check_mutable_borrow_field(&mut self, result_exp: &ExpData) {
+        result_exp.visit_pre_order(&mut |e| {
+            if let ExpData::Call(id, Operation::Borrow(ReferenceKind::Mutable), args) = &e {
+                if args.len() == 1 {
+                    if let ExpData::Call(_, Operation::Select(_, _, _), ref_targets) =
+                        args[0].as_ref()
+                    {
+                        if ref_targets.len() == 1
```

Comment:
Are there cases when `ref_targets.len()` will not be `1` here?

Reply:
In theory, it must be 1 but because it is also possible generate field selection operation in `post_process_body` where the length of args is not checked, I added this check.

Reply:
If it should be 1 (as well for the above `args.len() == 1`, maybe panic with ICE if not?